pkg.nix

```diff
@@ -0,0 +1,173 @@
+, version, src, libtexpdf-src
+, autoreconfHook
+, gitMinimal
+, pkg-config
+, jq
+, cargo
+, rustc
+, rustPlatform
+, makeWrapper
+, poppler_utils
+, harfbuzz
+, icu
+, fontconfig
+, lua
+, libiconv
+, darwin
+, makeFontsConf
+, gentium
+, runCommand
+}:
+
+let
+  luaEnv = lua.withPackages(ps: with ps; [
+    cassowary
+    cldr
+    cosmo
+    fluent
+    linenoise
+    loadkit
+    lpeg
+    lua-zlib
+    lua_cliargs
+    luaepnf
+    luaexpat
+    luafilesystem
+    luarepl
+    luasec
+    luasocket
+    luautf8
+    penlight
+    vstruct
+    # lua packages needed for testing
+    busted
+    luacheck
+  # NOTE: Add lua packages here, to change the luaEnv also read by `flake.nix`
+  ] ++ lib.optionals (lib.versionOlder lua.luaversion "5.2") [
+    bit32
+  ] ++ lib.optionals (lib.versionOlder lua.luaversion "5.3") [
+    compat53
+  ]);
+in stdenv.mkDerivation (finalAttrs: {
+  pname = "sile";
+  inherit version src;
+
+  preAutoreconf = ''
+    # Add the libtexpdf src instead of the git submodule. (From some reason
+    # without --no-preserve=mode flag, libtexpdf/ is unwriteable). As explained
+    # before, in Nixpkgs, we won't need to run these commands.
+    rm -rf ./libtexpdf
+    cp --no-preserve=mode -r ${libtexpdf-src} ./libtexpdf/
+    # pretend to be a tarball release so sile --version will not say `vUNKNOWN`.
+    echo ${finalAttrs.version} > .tarball-version
+  '';
+
+  nativeBuildInputs = [
+    autoreconfHook
+    gitMinimal
+    pkg-config
+    jq
+    cargo
+    rustc
+    rustPlatform.cargoSetupHook
+    poppler_utils
+    makeWrapper
+  ];
+  # In Nixpkgs, we don't copy the Cargo.lock file from the repo to Nixpkgs'
+  # repo, but we inherit src, and specify a hash (it is a fixed output
+  # derivation). `nix-update` and `nixpkgs-update` should be able to catch that
+  # hash and update it as well when performing updates.
+  cargoDeps = rustPlatform.importCargoLock {
+    lockFile = ./Cargo.lock;
+  };
+
+  buildInputs = [
+    luaEnv
+    harfbuzz
+    icu
+    fontconfig
+    libiconv
+  ] ++ lib.optionals stdenv.isDarwin [
+    darwin.apple_sdk.frameworks.AppKit
+  ];
+
+  configureFlags = [
+    "--with-system-luarocks"
+    # TODO: Explain this flag
+    "PDFINFO=false"
+    #"--with-manual" In Nixpkgs we add this flag, here its not important enough
+  ];
+
+  postPatch = ''
+    patchShebangs build-aux/*.sh
+  '' + lib.optionalString stdenv.isDarwin ''
+    sed -i -e 's|@import AppKit;|#import <AppKit/AppKit.h>|' src/macfonts.m
+  '';
+
+  NIX_LDFLAGS = lib.optionalString stdenv.isDarwin "-framework AppKit";
+
+  FONTCONFIG_FILE = makeFontsConf {
+    fontDirectories = [
+      gentium
+    ];
+  };
+
+  enableParallelBuilding = true;
+
+  preBuild = lib.optionalString stdenv.cc.isClang ''
+    substituteInPlace libtexpdf/dpxutil.c \
+      --replace "ASSERT(ht && ht->table && iter);" "ASSERT(ht && iter);"
+  '';
+
+  # remove forbidden references to $TMPDIR
+  preFixup = lib.optionalString stdenv.isLinux ''
```

Comment:
This one on the other hand a rather common thing you see in Nixpkgs, but we should verify it is still needed.

Reply:
How would I confirm if this is needed? Would the build fail with a warning about it if there was a problem? Because just dropping this block entirely seems to allow the Flake to continue running just fine. See #1941.